knotx-core/knotx-server/src/main/java/com/cognifide/knotx/server/KnotxServerVerticle.java

```diff
@@ -79,6 +79,7 @@ public void start(Future<Void> fut) throws IOException, URISyntaxException {
                                                         result -> {
                                                             TemplateEngineResponse engineResponse = new TemplateEngineResponse(result.body());
                                                             if (engineResponse.isSuccess()) {
+                                                                rewriteHeaders(request, request.headers());
```

Comment:
remove second parameter of rewriteHeaders() method. You already have request so headers can be taken inside the method.